### PR TITLE
Fix iterables not clearing previous values when updating with smaller lists

### LINKED ISSUE #106: Iterable cache isn't always set correctly
It seems there's a bug in the way iterables are rendered.

Currently this happens:
* Render 5 TemplateResult instances
* All 5 render correctly
* Render 1 TemplateResult instance (same template, new values)
* 1 TemplateResult is updated, 4 DOM nodes are [removed](https://github.com/PolymerLabs/lit-html/blob/9d6f581f77533cb47783d0271d6d968b1f7690de/src/lit-html.ts#L418)
* Render 5 TemplateResult instances
* `_previousValue` _still_ contains 5 instances, all 5 are [updated](https://github.com/PolymerLabs/lit-html/blob/9d6f581f77533cb47783d0271d6d968b1f7690de/src/lit-html.ts#L391)
* *4 new nodes are not appended*, so 1 is rendered and the remaining 4 which were previously removed never return

This seems to be because `_previousValue` is not re-computed when the clear() call runs (when extra DOM nodes are removed). So in future renders, lit-html believes 5 nodes still exist (they don't, only 1 is in the dom tree at this point).

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #106

### CODE REVIEW

File: src/lit-html.ts [hunk 1]
```diff
@@ -384,7 +384,7 @@ export class NodePart implements SinglePart {
 
     // Lets of keep track of how many items we stamped so we can clear leftover
```
Comment (cdata) [inline src/lit-html.ts]: `lets of => lets us`

File: src/lit-html.ts [hunk 2]
```diff
@@ -416,6 +416,7 @@ export class NodePart implements SinglePart {
       this._previousValue = undefined;
     } else if (partIndex < itemParts.length) {
       const lastPart = itemParts[partIndex - 1];
+      itemParts.length = partIndex;
```
Comment (cdata) [inline src/lit-html.ts]: 🎉 the first time I have ever seen someone set length on an `Array` is today 🎂

File: src/test/lit-html_test.ts [hunk 3]
```diff
@@ -451,8 +435,24 @@ suite('lit-html', () => {
         assert.equal(container.innerHTML, '<div>321</div>');
       });
 
+      test('updates arrays 2', () => {
```
Comment (cdata) [inline src/test/lit-html_test.ts]: Personal opinion: test titles like this ought to be more descriptive of what the test is actually doing.

File: src/lit-html.ts (hunk 1 above)
Comment (justinfagnani) [reply src/lit-html.ts]: fixed

File: src/lit-html.ts (hunk 2 above)
Comment (justinfagnani) [reply src/lit-html.ts]: first time for everything!

I added a comment

File: src/test/lit-html_test.ts (hunk 3 above)
Comment (justinfagnani) [reply src/test/lit-html_test.ts]: fixed

**Pull request merged**